frontend/src/Editor/Editor.jsx

```diff
@@ -161,6 +166,17 @@ class Editor extends React.Component {
     });
   };
 
+  // 1. When we receive an undoable action – we can always undo but cannot redo anymore.
```

Comment:
What is an undoable action? Switching between app versions?

Reply:
> > @sherfin94 added the fix
> 
> Hey @gondar00, looks like the crash still happens in certain scenarios:
> 
> 1. Add a button first
> 2. Then add a table
> 3. Click on table handle (so that inspector opens)
> 4. Undo twice
> 
> ![undo-issue](https://user-images.githubusercontent.com/20333972/145466214-aee345c7-1e83-42c3-90be-24dce82626c8.gif)

This I've added a check it seems to work fine now, thanks ^^

Reply:
Any change made to appDefinition is an undo/redo action 

resizing, deleting, moving the element on the canvas, add-edit component to canvas.

yes switching b/w app versions works too

Reply:
got it